sw/device/lib/testing/test_framework/check.h

```diff
@@ -258,4 +258,75 @@
     }                                              \
   } while (false)
 
+/**
+ * Evaluates a boolean expression for Ok or Error status, returning the false
+ * value.
+ *
+ * This macro is the `TRY` version for boolean expressions.
+ *
+ * This macro returns a `status_t` as opposed to `CHECK_*` macros that
+ * terminates the test.
+ *
+ * @param expr_ An expression that evaluates to `true` or `false`.
+ * @return The enclosed error value.
+ */
+#define ASSERT_TRUE(expr_, ...)     \
+  ({                                \
+    if (CHECK_IMPL(expr_)) {        \
+      dif_result_t res = kDifError; \
+      return INTO_STATUS(res);      \
+    }                               \
+  })
+
+/**
+ * Compare `num_items_` of `actual_` against `expected_` buffer.
+ *
+ * Prints differences between `actual_` and `expected_` before logging an error.
+ * Note, the differences between the actual and expected buffer values are
+ * logged via LOG_INFO _before_ the error is logged with LOG_ERROR, since by
+ * default DV simulations are configured to terminate upon the first error.
+ *
+ * This macro returns a `status_t` as opposed to `CHECK_*` macros that
+ * terminates the test.
+ *
+ * @param actual_ Buffer containing actual values.
+ * @param expected_ Buffer containing expected values.
+ * @param num_items_ Number of items to compare.
+ * @param ... Arguments to a LOG_* macro, which are evaluated if the check.
+ * @return The enclosed error value.
+ */
+#define ASSERT_ARRAYS_EQ(actual_, expected_, num_items_, ...)                  \
```

Comment:
I got you, it does make sense. But I thought we could keep the logging, but they would be ignored during manufacturing tests (we could have a mechanism in the OTTF to disable the logs in the tests and just send the `status_t` returned to the host) because the logs are very handy when debugging errors. Does that make sense?

Reply:
In manufacturing tests, we can log messages out of a test if we want, but they should NOT go over the OTTF console using a `LOG` (or `CHECK` macro that issues a `LOG`), they should be serialized to JSON using the ujson framework, so they are well formatted (e.g., defined by structs on both the device and host side), and can be passed through host-side gRPC services if need be.

We don't want to include switches into the OTTF that will change test behavior, we want to keep the tests self contained, the OTTF is simply a vehicle for booting tests in the proper bootstage / slot.

Reply:
Done